indexer/src/processors/kafka.rs

```diff
@@ -0,0 +1,75 @@
+use std::collections::HashMap;
+use std::time::Duration;
+
+use anyhow::anyhow;
+use rdkafka::config::ClientConfig;
+use rdkafka::producer::{FutureProducer, FutureRecord};
+
+use alloy::hex::ToHexExt;
+use indexer1::Processor;
+use multipool_storage::storage::parse_log;
+use serde_json::{json, to_value, Value};
+use sqlx::{Postgres, Transaction};
+
+pub struct KafkaEventProcessor {
+    topic: String,
+    producer: FutureProducer,
+}
+
+impl KafkaEventProcessor {
+    pub fn new(kafka_url: &str, topic: String) -> Self {
+        Self {
+            topic,
+            producer: ClientConfig::new()
+                .set("bootstrap.servers", kafka_url)
+                .create()
+                .expect("Cannot create kafka producer"),
+        }
+    }
+}
+
+impl Processor<Transaction<'static, Postgres>> for KafkaEventProcessor {
+    async fn process(
+        &mut self,
+        logs: &[indexer1::alloy::rpc::types::Log],
+        _transaction: &mut Transaction<'static, Postgres>,
+        _prev_saved_block: u64,
+        _new_saved_block: u64,
+        chain_id: u64,
+    ) -> anyhow::Result<()> {
+        let mut map = HashMap::new();
+        for l in logs.into_iter() {
+            map.entry(l.block_number.unwrap()).or_insert(vec![]).push(l);
+        }
+        for (block_number, logs) in map.into_iter() {
+            let logs: Value = logs
+                .into_iter()
+                .map(|log| {
+                    json!({
```

Comment:
why not using struct here and also just embedding log struct inside

Reply:
Within every block we can actually also put into the block body information about from_block that was scanned, then this can be used to work normally with storage's apply_events, it's also deduplication resistant as we just give info about the past and in case of a bigger range of blocks inserted everything will be consistent.